### ISSUE TITLE
Allow the sharing of a taskgraph cache across model runs in different workspaces

### ISSUE THREAD

**Issue by phargogh** [natcap/invest#7]
If I'm running a parameter study of a single model, let's say SDR, it would be really nice to be able to re-run the model in multiple workspaces (to keep the files separate), but only recompute the things that are needed.  It seems reasonable to pass in the taskgraph cache dir as an optional argument, and, if it's provided, the cache location should be reused.

**Comment by dcdenu4** [comment]
Bumping from the 3.9 milestone since it is not essential for the goal of the 3.9 release.

**Comment by dcdenu4** [comment]
I started working on this to the point of having it as an option in the Qt UI right below `n_workers` setting. I implemented it first in Habitat Quality and it seemed to be working as expected. The branch I was working on is here: https://github.com/dcdenu4/invest/tree/task/taskgraph-share-cache-across-workspace

In working on that problem I was trying to decide what `add_task` arguments would be necessary to allow `target` paths to NOT be included in the hashing, since the paths would be different given different workspace directories. This ultimately brought me to a conversation with Rich in which we discussed whether this might be outside of TaskGraphs purpose. 

The main takeaways from that conversation were:
- Hashing large rasters becomes a large and expensive tradeoff and might actually make things worse. This is in terms of deciding if two outputs are the same or not.
- Using a TG database as a catch all might not be a great idea.
- What is the problem a shared TG workspace is trying to solve? If it is to help organize and run a parameter study, maybe there is a better solution for that.
- One idea is to do all your pre-calculations up front and then set up your model runs and use those directly.
- `copy_duplicate_artifacts` argument might not have been such a great idea.

@richpsharp , I feel like I'm not doing a great job summarizing our conversation. Feel free to add context if you'd like.
